dash.el

```diff
@@ -1591,6 +1591,54 @@ and when that result is non-nil, through the next form, etc."
                    (--> ,result ,form))
                  ,@more))))
 
+(defmacro -cond-> (x &rest clauses)
+  "Conditionally thread X through CLAUSES.
+Threads x (via `->') through each form for which the
+corresponding test expression is true.  Note that, unlike cond
+branching, `-cond->' threading does not short circuit after the
+first true test expression."
+  (declare (debug (form body))
+           (indent 1))
+  (when (-> clauses length (% 2) (= 1))
+    (error "Wrong number of arguments"))
+  (-let* ((g (dash--match-make-source-symbol))
+          (steps (-map
+                  (-lambda ((test step))
+                    `(if ,test
+                         (-> ,g ,step)
+                       ,g))
+                  (-partition 2 clauses))))
+    `(-let* ((,g ,x)
+             ,@(-zip-lists (-cycle (list g))
+                           (butlast steps)))
+       ,@(if (null steps)
+             g
+           (last steps)))))
```

Comment:
If you build up `steps` in reverse, e.g. with a double `pop` and a `push` per clause, then you could use `car` and `cdr` instead of `last` and `butlast`, right?

Reply:
Sorry, I don't understand what you mean here. :(

Reply:
Perhaps I should make myself more clear:
I do understand the the general idea, how to use the built-in functions and why using them is a good idea.
What I'm not sure is the sequence of steps needed to achieve this.

Reply:
Does the following make sense?

```el
(defun dash--cond-thread (name thread expr clauses)
  "Implement macroexpansion of NAME, one of `-cond->' or `-cond->>'.
THREAD is the corresponding threading macro, one of `->' or
`->>'.  EXPR and CLAUSES are as in `-cond->'.  Signal
`wrong-number-of-arguments' if the number of CLAUSES is odd."
  (cond
   ((null clauses) expr)
   ((let ((len (length clauses)))
      (when (= (% len 2) 1)
        (signal 'wrong-number-of-arguments (list name (1+ len))))))
   ((let ((g (make-symbol "g"))
          binds)
      (while clauses
        (let ((test (pop clauses))
              (form (pop clauses)))
          (push `(,g (if ,test (,thread ,g ,form) ,g)) binds)))
      `(let* ((,g ,expr)
              ,@(nreverse (cdr binds)))
         ,(car (cdar binds)))))))
```

Then `-cond->` and `-cond->>` can be implemented as follows:

```el
(defmacro -cond-> (expr &rest clauses)
  (declare (indent 1))
  (dash--cond-thread #'-cond-> #'-> expr clauses))

(defmacro -cond->> (expr &rest clauses)
  (declare (indent 1))
  (dash--cond-thread #'-cond->> #'->> expr clauses))
```

Key differences:
- If there are no clauses, just return the first argument instead of signalling an error.
- `signal` takes two arguments.

Reply:
It's amazing!! I wish I had thought of this.

Reply:
If you think my suggestion is useful and clear enough, then feel free to incorporate it in your own code.

Reply:
Thank you!
I'm incorporating the part where steps are built whit two pop and a push, and the key differences listed above. But if I incorporated all the code, I'd be taking credit for your work. Perhaps you could update over this PR eventually.

Reply:
> But if I incorporated all the code, I'd be taking credit for your work.

But my work is already built on top of your work. ;) Also, my public suggestions are not copyrighted and I don't care about attribution.

> Perhaps you could update over this PR eventually.

Sure, if that's what you prefer, I'd be happy to oblige.

Reply:
> But my work is already built on top of your work. ;)

That's a nice way of looking at it.

> Sure, if that's what you prefer, I'd be happy to oblige.

IMO, this is the best option :)

Reply:
Cool, then I think we're just waiting for your CA process to complete, and we should be good to go.

BTW, if you rebase this branch off the latest `master` then the CI tests should stop wetting the bed. :)